### DISCUSSION TITLE
"App isn't responding" on Android, yet interactive

### DISCUSSION THREAD

**Discussion by tristanlabelle** [AvaloniaUI/Avalonia#9397] [Q&A]
My app is randomly showing the Android "App isn't responding" popup while I am fully able to interact with buttons and so on. When it starts happening, the popup shows repeatedly even if I can click "Wait" and keep using the UI until it pops again.

I have some asynchronous logic to crunch a few numbers on the thread pool and some `await Task.Delay(...)` logic on the UI thread that I use as a timer. I'm wondering if some of these are mistakenly detected as unresponsive threads by Android. Any ideas?

Edit: The ANR (Application Not Responding) event reads:

`Subject: Input dispatching timed out (29190e0 com.tristanlabelle.Solfear/crc6420235f80fe38c3a4.MainActivity (server) is not responding. Waited 5003ms for MotionEvent)`

**Comment by timunie** [comment]
Probably don't use Task.Delay for this. I'd use a DispatcherTimer to update numbers frequently.

**Reply by tristanlabelle** [reply]
My use case is easier to implement using `await Task.Delay`: I need to raise events at specific points in time while a media file is being played back. And how could this await interfere in any way with input event processing? I'm not blocking any UI event 🤔

**Reply by timunie** [reply]
Without seeing your code I cant judge. Probably it's something else. Can you post your entire function here?

**Reply by tristanlabelle** [reply]
Hmm, I'm having a hard time figuring out what makes this happen for a consistent. It might have to do with the debugger being attached. I would close this discussion but I don't see a way to do that.
